### ISSUE TITLE
Random Forest / XGBoost Classifier

### ISSUE THREAD

**Issue by Dhakshitha7** [Dhakshitha7/Bharat-Finance-Credit-Risk-Modeling#3]
# Define features and target
features = [col for col in df.columns if col not in ['timestamp', 'machine_id', 'failure']]
X = df[features]
y = df['failure']

# Train-test split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Normalize features
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Train classifier (Random Forest or XGBoost)
model_rf = RandomForestClassifier(n_estimators=100, random_state=42)
model_rf.fit(X_train_scaled, y_train)

# OR
# model_xgb = XGBClassifier(use_label_encoder=False, eval_metric='logloss')
# model_xgb.fit(X_train_scaled, y_train)

# Predict and evaluate
y_pred = model_rf.predict(X_test_scaled)
print(classification_report(y_test, y_pred))
